docs/11_search.md

```diff
@@ -0,0 +1,286 @@
+
+Use `workflow.GetInfo` to get current search attributes.
+
+### ContinueAsNew
+
+When performing a [ContinueAsNew](07_goclient/09_continueasnew), search attributes (and memo) will be carried over to the new run by default.
+
+## Query capabilities
+
+Query workflows by using a SQL-like where clause when [listing workflows from the CLI](08_cli#list-closed-or-open-workflow-executions) or using the list APIs ([Go](https://godoc.org/go.uber.org/cadence/client#Client), [Java](https://static.javadoc.io/com.uber.cadence/cadence-client/2.6.0/com/uber/cadence/WorkflowService.Iface.html#ListWorkflowExecutions-com.uber.cadence.ListWorkflowExecutionsRequest-))
+
+Note that you will only see workflows from one domain when querying.
+
+### Supported operators
+
+- AND, OR, ()
+- =, !=, >, >=, <, <=
+- IN
+- BETWEEN ... AND
+- ORDER BY
+
+### Default attributes
+
+These can be found using the CLI get-search-attr command or the GetSearchAttributes API:
+
+- CloseStatus = INT
+- CloseTime = INT
+- CustomBoolField = BOOL
+- CustomDatetimeField | DATETIME
+- CustomDoubleField = DOUBLE
+- CustomIntField = INT
+- CustomKeywordField = KEYWORD
+- CustomStringField = STRING
+- DomainID = KEYWORD
+- ExecutionTime = INT
+- HistoryLength = INT
+- RunID = KEYWORD
+- StartTime = INT
+- WorkflowID = KEYWORD
+- WorkflowType = KEYWORD
+
+CloseStatus, CloseTime, DomainID, ExecutionTime, HistoryLength, RunID, StartTime, WorkflowID, WorkflowType are reserved by Cadence and read-only
+
+- CloseStatus is a mapping of int to state:
+  - 0 = completed
+  - 1 = failed
+  - 2 = canceled
+  - 3 = terminated
+  - 4 = continuedasnew
+  - 5 = timedout
+- StartTime, CloseTime and ExecutionTime are stored as INT, but support queries using both EpochTime in nanoseconds, and string in RFC3339 format (ex. "2006-01-02T15:04:05+07:00")
+- CloseTime, CloseStatus, HistoryLength are only present in closed workflow
+- ExecutionTime is for Retry/Cron user to query workflow that will run in future
+
+To list only open workflows, add `CloseTime = missing` to the end of query.
+
+If you use retry or cron feature, to query workflows that will start execution in certain time range, you can add predicates on ExecutionTime. For example. `ExecutionTime > 2019-01-01T10:00:00-07:00`. Note that if predicates on ExecutionTime is included, only cron or workflow that needs retry will be returned.
+
+### General notes about queries
+
+- Pagesize is default to 1000, and can not be larger than 10k
+- Range query on Cadence timestamp (StartTime, CloseTime, ExecutionTime) cannot be larger than 9223372036854775807 (maxInt64 - 1001)
+- Query by time range will have 1ms resolution
+- Query column names are case sensitive
+- ListWorkflow may take longer time when retrieve large number of workflow (10M+)
```

Comment:
take longer time when retrieve large number of workflow -> take longer when retrieving a large number of workflows